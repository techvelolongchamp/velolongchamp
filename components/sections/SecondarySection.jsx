import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'

import Content from '../ui/Content'

import { SectionContainer, Title } from './SecondarySection.styled'

const SecondarySection = forwardRef(({ id, title, children, height }, ref) => (
  <SectionContainer name={id} height={height} ref={ref}>
    <Content>
      <Title>{title}</Title>
      {children}
    </Content>
  </SectionContainer>
))

SecondarySection.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  height: PropTypes.string,
}

export default SecondarySection
