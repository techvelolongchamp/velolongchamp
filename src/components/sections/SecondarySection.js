import React, { forwardRef } from 'react'
import styled from 'styled-components'

import Content from '../Content'

const SectionContainer = styled.section`
  min-height: ${({ theme, height }) => (height ? height : theme.sectionHeight)};
  background: ${({ theme }) => theme.colors.offWhite};
`

const Title = styled.h2`
  color: ${({ theme }) => theme.colors.brandPrimary};
  font-size: ${({ theme }) => theme.fonts.xlarge};
  margin: ${({ theme }) => theme.spacing.l} 0;
`

const SecondarySection = forwardRef(({ id, title, children, height }, ref) => (
  <SectionContainer ref={ref} id={id} height={height} className="anchor_tag">
    <Content>
      <Title>{title}</Title>
      {children}
    </Content>
  </SectionContainer>
))

export default SecondarySection
