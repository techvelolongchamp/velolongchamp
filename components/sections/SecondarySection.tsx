import React, { forwardRef } from 'react'

import Content from '../ui/Content'

import { SectionContainer, Title } from './SecondarySection.styled'

const SecondarySection = forwardRef<
  HTMLDivElement,
  { id?: string; title: string; height?: string; children: React.ReactNode }
>(({ id, title, children, height }, ref) => (
  <SectionContainer id={id} height={height} ref={ref}>
    <Content>
      <Title>{title}</Title>
      {children}
    </Content>
  </SectionContainer>
))

SecondarySection.displayName = 'SecondarySection'

export default SecondarySection
