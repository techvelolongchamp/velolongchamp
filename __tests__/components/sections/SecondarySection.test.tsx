import React, { useRef } from 'react'
import { RenderWithProvider as render } from '../../provider'

import SecondarySection from '../../../components/sections/SecondarySection'

const MockComponent = (props) => {
  const ref = useRef(null)

  return <SecondarySection ref={ref} {...props} />
}

describe('SecondarySection unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<MockComponent>Hello</MockComponent>)

    expect(container).toMatchSnapshot()
  })

  it('should render correctly - with id, title and height', () => {
    const { container, getByText } = render(
      <MockComponent height="300px" id="my-id" title="section-title">
        Hello
      </MockComponent>
    )
    const title = getByText('section-title')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
  })
})
