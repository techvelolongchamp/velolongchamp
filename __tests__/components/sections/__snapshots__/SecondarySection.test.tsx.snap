// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondarySection unit tests should render correctly - with id, title and height 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c0 {
  min-height: 300px;
  background: #f7f7F7;
}

.c2 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

<div>
  <section
    class="c0"
    height="300px"
    id="my-id"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        section-title
      </h2>
      Hello
    </div>
  </section>
</div>
`;

exports[`SecondarySection unit tests should render correctly 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c0 {
  min-height: 600px;
  background: #f7f7F7;
}

.c2 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      />
      Hello
    </div>
  </section>
</div>
`;
